src/lambdas/sendQueryResultsNotification/handler.ts

```diff
@@ -0,0 +1,67 @@
+import { EventBridgeEvent } from 'aws-lambda'
+import { getQueryByAthenaQueryId } from '../../sharedServices/dynamoDB/dynamoDBGet'
+import { updateZendeskTicketById } from '../../sharedServices/zendesk/updateZendeskTicket'
+import { AthenaEBEventDetails } from '../../types/athenaEBEventDetails'
+import { generateSecureDownloadHash } from './generateSecureDownloadHash'
+import { queueSendResultsReadyEmail } from './queueSendResultsReadyEmail'
+import { writeOutSecureDownloadRecord } from './writeOutSecureDownloadRecord'
+
+export const handler = async (
+  event: EventBridgeEvent<'Athena Query State Change', AthenaEBEventDetails>
+): Promise<void> => {
+  console.log('received event', JSON.stringify(event, null, 2))
+
+  const queryDetails = event.detail
+  const athenaQueryId = queryDetails.queryExecutionId
+
+  const requestData = await getQueryByAthenaQueryId(athenaQueryId)
+  const zendeskTicketId = requestData.requestInfo.zendeskId
+
+  await confirmQueryState(queryDetails, zendeskTicketId)
+
+  const recipientName = requestData.requestInfo.recipientName
+  const recipientEmail = requestData.requestInfo.recipientEmail
+  const downloadHash = generateSecureDownloadHash()
+
+  await writeOutSecureDownloadRecord(
+    athenaQueryId,
+    downloadHash,
+    zendeskTicketId
+  )
+
+  await queueSendResultsReadyEmail({
+    downloadHash,
+    recipientEmail,
+    recipientName,
+    zendeskTicketId
+  })
+}
+
+const confirmQueryState = async (
+  queryDetails: AthenaEBEventDetails,
+  zendeskId: string
+): Promise<void> => {
+  const queryState = queryDetails.currentState
+  switch (queryState) {
+    case 'SUCCEEDED': {
+      return
+    }
+
+    case 'QUEUED':
```

Comment:
For my benefit, can you explain why some cases are empty? Assume that means don't do anything... But what's the difference with the 'SUCCEEDED' case? Does the `return` make the handler stop at that point too?

Reply:
With the SUCCEEDED case, the return doesn't stop the handler but allows it to carry on.  Any case other than SUCCEEDED will result in an error and the handler ceasing to execute at that point

Reply:
Oh so it just uses the default and throws an error? Then do we we need these empty entries at all?

Reply:
@mabon-gds , I think @andylangridge might be right, we could just let the thing fall through to the default instead of specifically calling out QUEUED, and RUNNING, couldn't we? Just put `Function was called with unexpected state: ${queryState}. Ensure the template is configured correctly` in the default? We do need the succeeded case, because otherwise it'll error when the query succeeds, which is obviously no good.

Reply:
Right, this is done now